util/pkg/vfs/swiftfs.go

```diff
@@ -0,0 +1,453 @@
+/*
+Copyright 2016 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package vfs
+
+import (
+	"bytes"
+	"encoding/hex"
+	"fmt"
+	"net/http"
+	"os"
+	"path"
+	"path/filepath"
+	"runtime"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/go-ini/ini"
+	"github.com/golang/glog"
+	"github.com/gophercloud/gophercloud"
+	"github.com/gophercloud/gophercloud/openstack"
+	swiftcontainer "github.com/gophercloud/gophercloud/openstack/objectstorage/v1/containers"
+	swiftobject "github.com/gophercloud/gophercloud/openstack/objectstorage/v1/objects"
+	"github.com/gophercloud/gophercloud/pagination"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/kops/util/pkg/hashing"
+)
+
+func NewSwiftClient() (*gophercloud.ServiceClient, error) {
+	config := OpenstackConfig{}
+
+	authOption, err := config.GetCredential()
+	if err != nil {
+		return nil, err
+	}
+	provider, err := openstack.AuthenticatedClient(authOption)
+	if err != nil {
+		return nil, fmt.Errorf("error building openstack authenticated client: %v", err)
+	}
+	endpointOpt, err := config.GetServiceConfig("Swift")
+	if err != nil {
+		return nil, err
+	}
+
+	client, err := openstack.NewObjectStorageV1(provider, endpointOpt)
+	if err != nil {
+		return nil, fmt.Errorf("error building swift client: %v", err)
+	}
+	return client, nil
+}
+
+type OpenstackConfig struct {
+}
+
+func (_ OpenstackConfig) filename() (string, error) {
+	name := os.Getenv("OPENSTACK_CREDENTIAL_FILE")
+	if name != "" {
+		return name, nil
+	}
+
+	var homeDir string
+	if runtime.GOOS == "windows" {
+		homeDir = os.Getenv("USERPROFILE")
+	} else {
+		homeDir = os.Getenv("HOME")
+	}
+	if homeDir == "" {
+		return "", fmt.Errorf("can not find home directory")
+	}
+	return filepath.Join(homeDir, ".openstack", "config"), nil
```

Comment:
And I would probably `glog.V(2)` it here also then

Reply:
will update it!